### PR TITLE
[REF] web, project:  add hooks to calendar controler, fix css for filterPanel

### PR DESCRIPTION
**Pull request by bastvdn**
Description of the issue/feature this PR addresses:

- For the Planning app calendar we now need to be able to fetch specific fields from the filters record
we don't want to inherit the whole loadDynamicFilterSection method so we created an addFilterFields method
used to fetch specific fields from the record accroding to the filterInfo ()

- we also need to display specific text when creating a new record from the calendar view, for exemple in calendar 'New Shift' rather than 'New event' but we don't need to override the whole editRecord() method so we created a getter: editRecordDefaultDisplayText()

- For the css files, we lowered a bit the checkmark icon in the filters, and we added a new class that prevents the icons background to become invisible when the filter checkmark isn't changed

Task-3251698

See https://github.com/odoo/enterprise/pull/41304

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/calendar/calendar_model.js [hunk 1]
```diff
@@ -662,7 +670,7 @@ export class CalendarModel extends Model {
                     filters.push({
                         id: value,
                         [fieldName]: rawValue,
-                        colorIndex: record.colorIndex,
+                        ...this.addFilterFields(record, filterInfo),
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_model.js:673]: Cannot we change the `colorIndex` in record to have the one we want? 🤔 

I don't know if it is possible. 🙂

File: addons/web/static/src/views/calendar/calendar_controller.scss [hunk 2]
```diff
@@ -187,11 +187,19 @@ $o-cw-filter-avatar-size: 20px;
                     height: $o-cw-filter-avatar-size;
                     border-radius: 2px;
                     object-fit: cover;
+                    
+                    .fa-check {
+                        padding-top: 2px;
+                    }
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_controller.scss:193]: those changes should not be in planning since it is only in planning we display an icon instead of an image. 🤔

File: addons/web/static/src/views/calendar/calendar_controller.scss [hunk 3]
```diff
@@ -187,11 +187,19 @@ $o-cw-filter-avatar-size: 20px;
...
                 }
+
+                
             }
 
             input:not(:checked) + label .o_cw_filter_input_bg {
-                background: transparent !important;
+                &:not(.o_cw_filter_avatar) {
+                    background: transparent !important;
+                }
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_controller.scss:202]: same comment here

File: addons/web/static/src/views/calendar/calendar_controller.scss [hunk 4]
```diff
@@ -206,6 +214,9 @@ $o-cw-filter-avatar-size: 20px;
                 &.fa {
                     padding: 4px 3px;
                 }
+                &.fa-wrench {
+                    padding: 2px 2px;
+                }
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_controller.scss:219]: same comment here

File: addons/web/static/src/views/calendar/calendar_renderer.scss [hunk 5]
```diff
@@ -582,7 +582,7 @@
     }
 
     .o_cw_filter_color_#{$i - 1} {
-        .o_cw_filter_input_bg {
+        .o_cw_filter_input_bg:not(.no_filter_color) {
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_renderer.scss:585]: We should explain in the commit why we need to have this or we could also add it in planning instead.

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left some comments. 🙂 

Can you review your commits or squash your commits to have one for web and another one for project? Thanks in advance! 🙂

File: addons/web/static/src/views/calendar/calendar_controller.js [hunk 6]
```diff
@@ -165,7 +166,9 @@ export class CalendarController extends Component {
                         resModel: this.model.resModel,
                         resId: record.id || false,
                         context,
-                        title: record.id ? sprintf(_t("Open: %s"), record.title) : _t("New Event"),
+                        title: record.id
+                            ? sprintf(_t("Open: %s"), record.title)
+                            : this.editRecordDefaultDisplayText,
```
[the remark this answers is not in the document]
Comment (bastvdn) [reply addons/web/static/src/views/calendar/calendar_controller.js:184]: Done :+1:

File: addons/web/static/src/views/calendar/calendar_model.js (hunk 1 above)
Comment (bastvdn) [reply addons/web/static/src/views/calendar/calendar_model.js:673]: Mmh we still need the method because we also need the resource_type of the record ( so we know if we have to display a picture or a colored icon in the filterPanel). As for the colorIndex, according the the type of resource we don't fetch the same field to fill the colorIndex

File: addons/web/static/src/views/calendar/calendar_controller.scss (hunk 2 above)
Comment (bastvdn) [reply addons/web/static/src/views/calendar/calendar_controller.scss:199]: Nope in this case it's for the fa-check (checkmark that confirms that the filter is used) in all the current calendar view the checkmark is a little bit too high so i added the padding
image

File: addons/web/static/src/views/calendar/calendar_controller.scss (hunk 4 above)
Comment (bastvdn) [reply addons/web/static/src/views/calendar/calendar_controller.scss]: Yep there i can do it only in planning

File: addons/web/static/src/views/calendar/calendar_renderer.scss (hunk 5 above)
Comment (bastvdn) [reply addons/web/static/src/views/calendar/calendar_renderer.scss:620]: The fa-check icon that replace the avatar picture has the _o_cw_filter_input_bg_ class, but now one of it's parents has the o_cw_filter_color_{colorId} which wasn't the case previously their colorId was empty. Now that the fa-check has this color we need to make sure that there is a way to remove this background color or else the background of the check is colored as well.
I think it would be too heavy in term of code to cancel this from the planning calendar renderer, the class being inside of a css loop using specific colors constants. i'll explain it in the commit we'll see what they say

Comment (xavierbol) [reply addons/web/static/src/views/calendar/calendar_renderer.scss:620]: Okay thanks for the explanation. 🙂

File: addons/web/static/src/views/calendar/calendar_controller.scss (hunk 3 above)
Comment (xavierbol) [reply addons/web/static/src/views/calendar/calendar_controller.scss:206]: Okay, thanks for the explanation, we will see with the framework js then. 🙂

File: addons/web/static/src/views/calendar/calendar_controller.scss (hunk 2 above)
Comment (xavierbol) [reply addons/web/static/src/views/calendar/calendar_controller.scss:199]: Okay thanks for the explanation, let's see with framework js then. 🙂

File: addons/web/static/src/views/calendar/calendar_model.js (hunk 1 above)
Comment (xavierbol) [reply addons/web/static/src/views/calendar/calendar_model.js:673]: Indeed. 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

To be correct, the "web" commit should be before the other one. 🙂

I will let @odoo/rd-framework-js doing the final review. 🙂

File: addons/web/static/src/views/calendar/calendar_controller.scss (hunk 2 above)
Comment (adr-odoo) [reply addons/web/static/src/views/calendar/calendar_controller.scss:199]: Could you use ´align-items-center´ on ´.o_cw_filter_input_bg.o_beside_avatar´ OR Remove the width and height.

File: addons/web/static/src/views/calendar/calendar_controller.js [hunk 7]
```diff
@@ -116,6 +116,10 @@ export class CalendarController extends Component {
         return this.props.className;
     }
 
+    get editRecordDefaultDisplayText() {
+        return _t("New Event");
```
Comment (xavierbol) [inline addons/web/static/src/views/calendar/calendar_controller.js:120]: We could add a js test in web to test that method when we override it in custom view.

We could do something like that:
https://github.com/odoo/odoo/pull/120094/files#diff-89c6ed60f280620223e211f7c87cd1a1824696866a6eab08688d620153d781e7R4245

**Review by xavierbol — COMMENTED** [review]
I think we have to add js test before pinging again the framework JS. 🙈

**Comment by xavierbol** [comment]
Hello @odoo/rd-framework-js 

Can we have a review when you have time ? 

Thanks in advance! 

Have a nice weekend !

File: addons/web/static/src/views/calendar/calendar_model.js
Comment (mcm-odoo) [inline addons/web/static/src/views/calendar/calendar_model.js:1]: It seems that the diff in this file is useless. There is no (not yet?) override project side.

Comment (xavierbol) [reply addons/web/static/src/views/calendar/calendar_model.js:1]: Actually, it is used in planning module 🙂 

https://github.com/odoo/enterprise/pull/41304/files#diff-93df982529187d8a1c2faf011e7dd5adab79f9a0162b185451b72c7f4e084b1fR22

Comment (mcm-odoo) [reply addons/web/static/src/views/calendar/calendar_model.js:1]: Ok then, it's good for me :smile:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks again for your work. 🙂 

Thanks @mcm-odoo for your review. 🙂 

robodoo rebase-ff r+

**Comment by d-fence** [comment]
robodoo r-
[Seems to break the staging](https://runbot.odoo.com/runbot/build/50023144)

File: addons/project/static/src/views/project_task_calendar/project_task_calendar_controller.js [hunk 8]
```diff
@@ -3,6 +3,7 @@
 import { CalendarController } from "@web/views/calendar/calendar_controller";
 import { ProjectTaskCalendarFilterPanel } from "./project_task_calendar_filter_panel/project_task_calendar_filter_panel";
 import { DeleteSubtasksConfirmationDialog } from "@project/components/delete_subtasks_confirmation_dialog/delete_subtasks_confirmation_dialog";
+import { _t } from "@web/core/l10n/translation";
```
Comment (xavierbol) [inline addons/project/static/src/views/project_task_calendar/project_task_calendar_controller.js:6]: Hello @bastvdn ,

Can you rebase your branches a last time and remove that line because it will be imported by another commit? 🙂 

Sorry about that. 🙂

**Review by xavierbol — APPROVED** [review]
Thanks @bastvdn 🙂 

robodoo r+
